presto-spark-base/src/main/java/com/facebook/presto/spark/execution/HttpNativeExecutionTaskInfoFetcher.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.execution;
+
+import com.facebook.airlift.concurrent.SetThreadName;
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.execution.TaskId;
+import com.facebook.presto.execution.TaskInfo;
+import com.facebook.presto.server.smile.BaseResponse;
+import com.facebook.presto.spark.execution.http.PrestoSparkHttpWorkerClient;
+import com.google.common.util.concurrent.FutureCallback;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import io.airlift.units.Duration;
+
+import javax.annotation.concurrent.GuardedBy;
+
+import java.util.concurrent.Executor;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This class helps to fetch TaskInfo for a native task through HTTP communications with a Presto worker. Upon callling start(), object of this class will continously poll TaskInfo
+ * from Presto worker and update its internal TaskInfo buffer. Caller is responsible for retrieving updated TaskInfo by calling getTaskInfo() method. Caller is also responsible for
+ * calling stop() to release resource when this fetcher is no longer needed.
+ */
```

Comment:
nit:
- there are typos in callling and continously
- break lines with your own judgement.
- use `{@link TaskInfo}` with `import com.facebook.presto.execution.TaskInfo;` to do x-ref
```java
/**
 * This class helps to fetch {@link TaskInfo} for a native task through HTTP communications with a Presto worker.
 * Upon calling start(), object of this class will continuously poll {@link TaskInfo}
 * from Presto worker and update its internal {@link TaskInfo} buffer.
 * Caller is responsible for retrieving updated {@link TaskInfo} by calling getTaskInfo() method.
 * Caller is also responsible for calling stop() to release resource when this fetcher is no longer needed.
 */
```